lib/money/currency.rb

```diff
@@ -34,18 +34,17 @@ def initialize(method, currency, attribute)
     class UnknownCurrency < ArgumentError; end
 
     class << self
-      alias_method :original_new, :new
       def new(id)
         id = id.to_s.downcase
         unless stringified_keys.include?(id)
           raise UnknownCurrency, "Unknown currency '#{id}'"
         end
 
-        @@mutex.synchronize { instances[id] } || super
+        @@mutex.synchronize { instances[id] ||= super }
       end
 
       def instances
-        @instances ||= Hash.new { |h, k| h[k] = original_new(k) }
+        @instances ||= Hash.new
```

Comment:
I've missed: my variant won't be thread-safe, while this is lazy. Init could be moved out to class definition, and replaced with attr_reader.